docs/concepts/executor/executor-in-flow.md

```diff
@@ -0,0 +1,143 @@
+
+These can be provided to the Executor via the Python or {ref}`YAML API <executor-yaml-spec>`.
+
+(executor-metas)=
+### `metas`
+
+An Executor object contains `metas` as an attribute when loaded from the Flow. It is of [`SimpleNamespace`](https://docs.python.org/3/library/types.html#types.SimpleNamespace) type and contains some key-value information. 
+
+The list of the `metas` are:
+
+- `name`: Name given to the Executor;
+- `description`: Description of the Executor (optional, reserved for future-use in auto-docs);
+
+
+These can be provided to the Executor via the Python or {ref}`YAML API <executor-yaml-spec>`.
+
+### `runtime_args`
+
+By default, an Executor object contains `runtime_args` as an attribute when loaded from the Flow. It is of [`SimpleNamespace`](https://docs.python.org/3/library/types.html#types.SimpleNamespace) type and contains information in key-value format. 
+As the name suggests, `runtime_args` are dynamically determined during runtime, meaning that you don't know the value before running the Executor. These values are often related to the system/network environment around the Executor, and less about the Executor itself, like `shard_id` and `replicas`. They are usually set with the {meth}`~jina.orchestrate.flow.base.Flow.add` method.
+
+The list of the `runtime_args` is:
+
+- `name`: Name given to the Executor. This is dynamically adapted from the `name` in `metas` and depends on some additional arguments like `shard_id`. 
+- `replicas`: Number of {ref}`replicas <replicate-executors>` of the same Executor deployed with the Flow.
+- `shards`: Number of {ref}`shards <partition-data-by-using-shards>` of the same Executor deployed with the Flow.
+- `shard_id`: Identifier of the `shard` corresponding to the given Executor instance.
+- `workspace`: Path to be used by the Executor. Note that the actual workspace directory used by the Executor is obtained by appending `'/<executor_name>/<shard_id>/'` to this value.
+- `py_modules`: Python package path e.g. `foo.bar.package.module` or file path to the modules needed to import the Executor. This is another way to pass `py-modules` to the Executor from the Flow
+
+You **cannot** provide these through any API. They are generated by the Flow orchestration.
+
+## Call a Flow from an Executor
+
+To call other another Jina Flow using `Client` from an `Executor`, you also need to use `async def` and async Client.
+
+```python
+from jina import Client, Executor, requests, DocumentArray
+
+
+class DummyExecutor(Executor):
+    c = Client(host='grpc://0.0.0.0:51234', asyncio=True)
+
+    @requests
+    async def process(self, docs: DocumentArray, **kwargs):
+        self.c.post('/', docs)
+```
+
+## Serve
+
+Both served and shared Executors can be used as part of a Flow, by adding them as an {ref}`external Executor <external-executors>`.
```

Comment:
I believe we need to update this narrative @JoanFM .
Since now even worker runtime can communicate with Client, I guess users shouldn't care about gateway anymore